### DISCUSSION TITLE
DECK_CORE: Deck 0 test [FAIL].

### DISCUSSION THREAD

**Discussion by Residualstress** [bitcraze/discussions#1542] [Q&A] [answered]
Hi, I met problem `DECK_CORE: Deck 0 test [FAIL]` when powered the Crazyflie 2.0. Here is the detail:

When I powered Crazyflie 2.0 + Battery holder + Flow deck, the cfclient console output:
```
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:e527895e8c6e (2023.11) CLEAN
SYS: I am 0x2039365637565007004B0030 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow2 for deck 0
ZR2: Z-down sensor [FAIL]
PMW: Motion chip id: 0x49:0xB6
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CF21)
EEPROM: I2C connection [FAIL].
SYS: configblock [FAIL]
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [FAIL].
SYS: deck [FAIL]
```
When I powered Crazyflie 2.0 + Battery holder, the console output:
```
SYS: I am 0x2039365637565007004B0030 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 0 deck(s) found
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CF21)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 20496 bytes
STAB: Starting stabilizer loop
SUP: Can not fly
SUP: Ready to fly
```
I was confirmed that the Battery holder and Flow deck work are in good condition since they work well in another Crazyflie 2.0. And the firmware version is the latest.

So where could something have gone wrong? My guess is that the Crazyflie 2.0 has a loose circuit connection on it.

**Comment by ataffanel** [comment, marked as answer]
It looks like something is wrong with the I2C bus when connecting the flow deck: the internal eeprom I2C is connected on the same I2C bus as the flow deck on the deck connector, and when connecting the flow deck, connection to the EEPROM seems to fail. So this suggest that the deck is either shorting or at least affecting the bus in a bad way.

You can check the state of your flow deck to see if there is any damage or shorts on it. If this is a new-ish board and you cannot find any obvious problem, you can send a mail to support _at_ bitcraze.io and we will fix a replacement.
